Assets/EnemyAI.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyAI : MonoBehaviour
+{
+    public Transform chaseTarget;
+    public float moveSpeed = 5f;
+    private Rigidbody2D rb;
+    public float health;
+    public float damageOverTime = 1f;
+    private float damageTimer;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        rb = GetComponent<Rigidbody2D>();
+        damageTimer = damageOverTime;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        Vector3 targetDirection = chaseTarget.position - transform.position;
+        
+        transform.position = Vector2.MoveTowards(transform.position, chaseTarget.position, moveSpeed * Time.deltaTime);
+
+        if (damageTimer > 0)
```

Comment:
This code might be better suited in a timer coroutine to count down the time rather than in a framerate-dependent loop (could get odd time warping or inaccurate skipping). So, instead of counting it down here, we can instead start a coroutine and await its completion before the damage can be done again in the collision stay function below.